docs/gettinginvolved/plugins.rst

```diff
@@ -0,0 +1,93 @@
+=======================
+Dataset Backend Plugins
+=======================
+
+Flocker supports pluggable storage backends; this document will teach you how to implement such a backend, with a particular focus on block device backends.
+
+
+Block device backends
+=====================
+
+Flocker implements generic logic for network-based block device storage.
+Examples of such storage include Amazon EBS and EMC ScaleIO.
+If you wish to support a block device that is not supported by Flocker or an existing plugin you can implement this support yourself.
+
+In order to do so you must write a Python 2.7 library providing a class implementing the `flocker.node.agents.blockdevice.IBlockDeviceAPI <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/blockdevice.py>`_ interface.
```

Comment:
:question: Maybe a follow up, but it might be nice to use some of this interface in the docs with [literalinclude](http://sphinx-doc.org/markup/code.html#includes), e.g.

```rst
.. literalinclude:: ../../flocker/node/agents/blockdevice.py
   :pyobject: IBlockDeviceAPI
```

This will also highlight that the doc needs changing if the interface moves or is renamed for example.